platform-grpc-service-framework/src/main/java/org/hypertrace/core/serviceframework/grpc/GrpcPlatformServiceContainer.java

```diff
@@ -113,8 +168,32 @@ protected String getInProcessServerName() {
     return this.getServiceName();
   }
 
-  protected abstract Collection<GrpcPlatformServiceFactory> getServiceFactories();
+  protected ScheduledExecutorService buildTaskExecutor(int taskCount) {
+    // Between 1-4 threads
+    return Executors.newScheduledThreadPool(Math.max(1, Math.min(taskCount, 4)));
```

Comment:
this does not need to be configurable?

Reply:
I don't think so, but I left it protected so it could be overridden if needed. This wouldn't need runtime config anyway, would be a function of how many tasks are registered.